plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/impl/RenameElementHandler.java

```diff
@@ -39,22 +44,46 @@
 import org.yakindu.sct.refactoring.handlers.AbstractRefactoringHandler;
 import org.yakindu.sct.refactoring.refactor.AbstractRefactoring;
 import org.yakindu.sct.refactoring.refactor.impl.RenameRefactoring;
+import org.eclipse.xtext.diagnostics.Severity;
 
 import com.google.common.collect.Lists;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
 /**
  * Handler for {@link RenameRefactoring}.
  * 
  * @author thomas kutz - Initial contribution and API
  * 
  */
 public class RenameElementHandler extends AbstractRefactoringHandler<NamedElement> {
+	
+	@Inject
+	private IResourceValidator validator;
 
+	
+	public RenameElementHandler() {
+		Guice.createInjector().injectMembers(this);
+	}
+
+	
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
 		NamedElement element = refactoring.getContextObject();
 		if (element != null) {
 			IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
-			InputDialog dialog = new InputDialog(window.getShell(), "Rename..", "Please enter new name:", element.getName(), new NameUniquenessValidator(element));
+			
+
+			List<Issue> issues = validator.validate(element.eResource(), CheckMode.NORMAL_AND_FAST, CancelIndicator.NullImpl);
+			Stream<Issue> errors = issues.stream().filter(issue -> issue.getSeverity() == Severity.ERROR);
+			
+			String message = "";
+			if (errors.count() > 0) {
+				message = "File contains errors!!\n\n";
```

Comment:
There should be a status bar indicating that the file contains errors and the renaming may be unsuccessful. Add a warning icon. Maybe you can use / reuse parts of the TitleAreaDialog 
http://www.vogella.com/tutorials/EclipseDialogs/article.html